types/app.d.ts

```diff
@@ -259,7 +259,7 @@ declare namespace Realm {
          * All authenticated users.
          */
         readonly allUsers: Readonly<
-            User<FunctionsFactoryType, CustomDataType>[]
+            Record<string, User<FunctionsFactoryType, CustomDataType>>
```

Comment:
this needs a changelog entry in Realm JS

Reply:
@steffenagger suggested this one
* Fixed TS declaration for `app.allUsers` to `Record<string, User>` instead of `Array<User>` ([#3346](https://github.com/realm/realm-js/pull/3346))